### PR TITLE
FE: Profile Settings Skeleton #45

### PR DESCRIPTION
**Pull request by jstyle5**
### What this PR does (required):
- Issue Link: https://github.com/hatchways/team-breadsticks/issues/45
- Created a functional skeleton for the profile settings page
  - sidebar: Edit profile, Profile Photo, Payment, Security, Settings
  - main content (just the empty paper component)

#### Desktop version (After logging in, click UserAvatar in NavBar and click ```PROFILE``` in Dropdown, URL: ```http://localhost:3000/profile```)
Screen Shot 2021-09-19 at 19 31 54
---

#### Desktop version (After clicking ```Edit Profile``` menu in Sidebar, URL: ```http://localhost:3000/profile/edit```)
Screen Shot 2021-09-19 at 19 40 43
---

#### Desktop version (After clicking ```Profile Photo``` menu in Sidebar, URL: ```http://localhost:3000/profile/photo```)
Screen Shot 2021-09-19 at 19 42 30
---

#### Desktop version (After clicking ```Payment``` menu in Sidebar, URL: ```http://localhost:3000/profile/payment```)
Screen Shot 2021-09-19 at 19 45 14
---

#### Desktop version (After clicking ```Security``` menu in Sidebar, URL: ```http://localhost:3000/profile/security```)
Screen Shot 2021-09-19 at 19 45 19
---

#### Desktop version (After clicking ```Settings``` menu in Sidebar, URL: ```http://localhost:3000/profile/settings```)
Screen Shot 2021-09-19 at 19 45 24
---

#### Mobile version (After logging in, click UserAvatar in NavBar and click ```PROFILE``` in Dropdown, URL: ```http://localhost:3000/profile```)
Screen Shot 2021-09-19 at 19 47 26
---

#### Mobile version (After clicking ```Profile Menus``` below UserAvatar in NavBar to open SwipeableDrawer, URL: ```http://localhost:3000/profile```)
Screen Shot 2021-09-19 at 19 51 12
---

#### Mobile version (After clicking ```Edit Profile``` in SwipeableDrawer, URL: ```http://localhost:3000/profile/edit```)
Screen Shot 2021-09-19 at 19 52 12
---

#### Mobile version (After clicking ```Profile Menus``` below UserAvatar in NavBar to open SwipeableDrawer and choose other menus, URL: ```http://localhost:3000/profile/edit```)
Screen Shot 2021-09-19 at 19 54 46
---

#### Mobile version (After clicking ```Profile Photo``` below UserAvatar in NavBar to open SwipeableDrawer, URL: ```http://localhost:3000/profile/photo```)
Screen Shot 2021-09-19 at 19 56 09
---

#### Mobile version (After clicking ```Profile Menus``` below UserAvatar in NavBar to open SwipeableDrawer and choose other menus, URL: ```http://localhost:3000/profile/photo```)
Screen Shot 2021-09-19 at 19 59 39
---

### Any information needed to test this feature (required):
- Assume that the ```loggedInUser``` is a pet owner to display ```My Sitters``` link on the navbar
- (e.g., ```loggedInUser.isSitter = false // hardcoded```)
- If the ```loggedInUser``` is a pet sitter, then ```My Jobs``` link will be shown on the navbar instead of ```My Sitters```

#### Desktop version
- After logging in, click UserAvatar in NavBar
- Click ```PROFILE``` in Dropdown
- Click menus in SideBar (e.g., ```Edit Profile```, ```Profile Photo```, ```Payment```, ```Security```, ```Settings```)
- Contents will be displayed on the right side of the screen based on the menu that you click

#### Mobile version
- After logging in, click UserAvatar in NavBar
- Click ```PROFILE``` in Dropdown
- Click ```Profile Menus``` below UserAvatar or swipe from the left side of the screen to the right side of the screen to open SwipeableDrawer
- Click menus in SwipeableDrawer (e.g., ```Edit Profile```, ```Profile Photo```, ```Payment```, ```Security```, ```Settings```)

### Any issues with the current functionality (optional):
- N/A

### CODE REVIEW

**Review by ezeilo-su — COMMENTED** [review]
Your UI looks cool. I like those screenshots in the PR. Impressive that you made it responsive.

**Comment by jstyle5** [comment]
> Your UI looks cool. I like those screenshots in the PR. Impressive that you made it responsive.

Thank you :)

File: client/src/App.tsx [hunk 1]
```diff
@@ -23,33 +28,80 @@ function App(): JSX.Element {
...
                 </Route>
+
                 <Route exact path="/profile">
                   <Dashboard>
                     <Profile />
                   </Dashboard>
                 </Route>
+
+                {/* TODO : Refactoring all routes? */}
+                <Route exact path="/profile/edit">
```
Comment (rajivtitus) [inline client/src/App.tsx:47]: Consider the usage of path parameters here.

Eg: `<Route to='/profile/:setting' component={Profile}`

File: client/src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
   return (
-    <Grid className={classes.profile}>
-      <Box>
-        <Typography component="h5" variant="h5" align="center">
-          Profile Page Template
-        </Typography>
-      </Box>
+    <Grid container className={classes.profile}>
+      <ProfileSideBar />
+      {children ? (
```
Comment (rajivtitus) [inline client/src/pages/Profile/Profile.tsx:16]: As opposed to displaying the children here, you can extract the 'path' name from the url and conditionally render the component which matches the path name.

File: client/src/pages/Profile/ProfileSideBar/ProfileSideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const sideBarMenus = [
+  { menuItem: 'Edit Profile', linkTo: '/profile/edit' },
+  { menuItem: 'Profile Photo', linkTo: '/profile/photo' },
+  { menuItem: 'Payment', linkTo: '/profile/payment' },
+  { menuItem: 'Security', linkTo: '/profile/security' },
+  { menuItem: 'Settings', linkTo: '/profile/settings' },
+];
+
+const ProfileSideBar = (): JSX.Element => {
```
Comment (rajivtitus) [inline client/src/pages/Profile/ProfileSideBar/ProfileSideBar.tsx:20]: Good work on this component!

**Review by rajivtitus — COMMENTED** [review]
Good use of all the technologies! Great work on the UI and adding mobile responsiveness.
I've added a couple of comments, please have a look and address. Thanks

File: client/src/App.tsx (hunk 1 above)
Comment (jstyle5) [reply client/src/App.tsx:35]: Applied

File: client/src/pages/Profile/Profile.tsx (hunk 2 above)
Comment (jstyle5) [reply client/src/pages/Profile/Profile.tsx]: Applied.
I extracted the ```pathname``` from ```useLocation()``` and rendered components conditionally with ```switch``` statement. It seems to work fine as I intended. But, I am not sure whether if I fixed it nicely or not.

**Comment by jstyle5** [comment]
> Good use of all the technologies! Great work on the UI and adding mobile responsiveness.
> I've added a couple of comments, please have a look and address. Thanks

Thank you :)

File: client/src/pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -1,22 +1,43 @@
...
+import Security from './Security/Security';
+import Settings from './Settings/Settings';
 
 const Profile = (): JSX.Element => {
-  // TODO: this will be used in the future
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const classes = useStyles();
+  const location = useLocation();
+
+  const content = () => {
```
Comment (rajivtitus) [inline client/src/pages/Profile/Profile.tsx:17]: 👍🏽

**Review by rajivtitus — APPROVED** [review]
Great work translating my comments into code 💯

**Comment by jstyle5** [comment]
> Great work making those changes - 👍

Thank you :)

**Comment by jstyle5** [comment]
> Great work translating my comments into code 💯

Thank you :)

**Pull request merged**